system/lib/wasmfs/wasmfs.cpp

```diff
@@ -63,9 +64,36 @@ extern "C" void wasmfs_flush(void) {
   // Flush musl libc streams.
   fflush(0);
 
-  // Flush our own streams. TODO: flush all backends.
+  // Flush our own streams.
   (void)SpecialFiles::getStdout()->locked().flush();
   (void)SpecialFiles::getStderr()->locked().flush();
+
+  std::stack<std::shared_ptr<Directory>> toFlush;
+  toFlush.push(wasmFS.getRootDirectory());
+  
+  while(!toFlush.empty()) {
+    auto dir = toFlush.top();
+    toFlush.pop();
+    
+    auto lockedDir = dir->locked();
+    Directory::MaybeEntries entries = lockedDir.getEntries();
+    if (int err = entries.getError()) {
+      std::string errorMessage = "Non-fatal error code " + std::to_string(err) + " while flushing directory: " + lockedDir.getName(dir);
```

Comment:
I think this error logging added a lot of code size because of the string operations. Replacing this with `emscripten_console_error("Non-fatal error while flushing filesystem.")` changed the code size to `55968`, which is still up from the original `52346` but better than `58926` with the extra logging information. Trying to log the extra information with `emscripten_console_errorf` introduces an even bigger code size increase (Increased to `65656`). Does the extra information warrant the code size increase?

Reply:
Definitely in a release build we wouldn't want that info, I think. Any error reporting can be guarded with that ifdef I think we mentioned in a meeting,
```cpp
#ifndef NDEBUG
  // debug code here
#endif
```